test/test-listing-guni.spec.ts

```diff
@@ -0,0 +1,233 @@
+    await (await gov.queue(proposal)).wait();
+    let proposalState = await gov.getProposalState(proposal);
+    expect(proposalState).to.be.equal(5);
+    await increaseTime(86400 + 10);
+    snapshotId = await evmSnapshot();
+  });
+
+  it('Should list correctly an asset: borrow off, collateral on, stable borrow off', async () => {
+    await (await gov.execute(proposal)).wait();
+    
+    const proposalState = await gov.getProposalState(proposal);
+    expect(proposalState).to.be.equal(7);
+    const {
+      configuration: { data },
+      aTokenAddress,
+      stableDebtTokenAddress,
+      variableDebtTokenAddress,
+    } = await pool.getReserveData(GUNI1);
+    const poolData = parsePoolData(data);
+    expect(poolData).to.be.eql({
+      reserveFactor: '1000',
+      reserved: '0',
+      stableRateEnabled: '0',
+      borrowingEnabled: '0',
+      reserveFrozen: '0',
+      reserveActive: '1',
+      decimals: '18',
+      liquidityBonus: '11500',
+      LiquidityThreshold: '7000',
+      LTV: '6000',
+    });
+
+    // preparing for tests.
+    console.log("prepare for tests...")
+    aGuni = (await ethers.getContractAt('IERC20', aTokenAddress, proposer)) as IERC20;
+    stableDebt = (await ethers.getContractAt('IERC20', stableDebtTokenAddress, proposer)) as IERC20;
+    variableDebt = (await ethers.getContractAt(
+      'IERC20',
+      variableDebtTokenAddress,
+      proposer
+    )) as IERC20;
+    await (await guni.connect(guniHolder).approve(pool.address, parseEther('200000'))).wait();
+    await (await aave.connect(proposer).approve(pool.address, parseEther('200000'))).wait();
+
+    // AAVE deposit by proposer
+    /*console.log("aave deposit by proposer")
+    await (await pool.deposit(aave.address, parseEther('10'), proposer.address, 0)).wait();*/
+
+    // GUNI deposit by guni holder
+    console.log("guni deposit by g uni holder")
+    const depositedAmount = parseEther('3');
+    await (
+      await pool.connect(guniHolder).deposit(guni.address, depositedAmount, GUNI_HOLDER, 0)
+    ).wait();
+    expect(await aGuni.balanceOf(GUNI_HOLDER)).to.gte(depositedAmount.sub(1));
+    expect(await aGuni.balanceOf(GUNI_HOLDER)).to.lte(depositedAmount.add(1));
+
+    // G-UNI holder able to borrow DAI against G-UNI
+    console.log("guni holder borrows dai")
+    await (
+      await pool.connect(guniHolder).borrow(dai.address, parseEther('1'), 2, 0, GUNI_HOLDER)
+    ).wait();
+
+    // proposer NOT able to borrow G-UNI
+    const borrowAmount = parseEther('10').div(decimalMultiplier);
+    await expect(
+      pool.connect(proposer).borrow(guni.address, borrowAmount, 2, 0, proposer.address)
+    ).to.be.reverted;
+
+    await expect(
+      pool.borrow(guni.address, borrowAmount, 1, 0, proposer.address)
+    ).to.be.reverted;
+    increaseTime(40000);
+  });
+
+  it("Oracle should return a non zero G-UNI price", async () => {
+    expect(await oracle.getAssetPrice(GUNI1)).to.be.gt('0')
```

Comment:
check oracle addition for all the assets getting listed please

Reply:
done